### DISCUSSION TITLE
How to do similarity_search with specify field

### DISCUSSION THREAD

**Discussion by qqqqqwwwweeerr1gmail** [milvus-io/milvus#28193] [Q&A and General discussion]
embeddings = llms_consturctor.azure_openai_embeddings()
collection_name = 'test_col_10006'
question = "gggggggggggggggggg"
db_name = 'knowledge_repo_test'

vectordb = Milvus.from_texts(
        [],
        embeddings,
        collection_name=collection_name,
        connection_args={"host": "******", "port": "19530","db_name":db_name},
    )

docs = vectordb.similarity_search(
    question,
    k=3,
    filter={"paper_title":"rrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrr"}
)

or this code :

    db = Milvus.from_documents(
        [],
        embeddings,
        collection_name=collection_name,
        connection_args={"host": "10.5.121.43", "port": "19530","db_name":db_name},
    )
    retriever = db.as_retriever(search_type="similarity", search_kwargs={"k": sim_k, "filter":{"paper_title":"rrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrr"}})

I run this code but the filter not worked . How to fix similarity_search with specify field named 'paper_title', the search kages is vallyed

**Comment by yhmo** [comment]
The definition of langchain.vectorstores.milvus.similarity_search()
```
 def similarity_search(
        self,
        query: str,
        k: int = 4,
        param: Optional[dict] = None,
        expr: Optional[str] = None,
        timeout: Optional[int] = None,
        **kwargs: Any,
    ) -> List[Document]:
        """Perform a similarity search against the query string.

        Args:
            query (str): The text to search.
            k (int, optional): How many results to return. Defaults to 4.
            param (dict, optional): The search params for the index type.
                Defaults to None.
            expr (str, optional): Filtering expression. Defaults to None.
            timeout (int, optional): How long to wait before timeout error.
                Defaults to None.
            kwargs: Collection.search() keyword arguments.

        Returns:
            List[Document]: Document results for search.
        """
```

We can see there is a parameter "expr" passed to milvus.
If you know the field name then you can use [milvus expression](https://milvus.io/docs/boolean.md) to do filtering.
